src/gui/pointcloud/qgspointcloudrendererpropertieswidget.cpp

```diff
@@ -124,13 +125,45 @@ QgsPointCloudRendererPropertiesWidget::QgsPointCloudRendererPropertiesWidget( Qg
   connect( mHorizontalTriangleThresholdSpinBox, qOverload<double>( &QgsDoubleSpinBox::valueChanged ), this, &QgsPointCloudRendererPropertiesWidget::emitWidgetChanged );
   connect( mHorizontalTriangleUnitWidget, &QgsUnitSelectionWidget::changed, this, &QgsPointCloudRendererPropertiesWidget::emitWidgetChanged );
 
-  // show label options only for virtual point clouds
-  bool showLabelOptions = !mLayer->dataProvider()->subIndexes().isEmpty();
-  mLabels->setVisible( showLabelOptions );
-  mLabelOptions->setVisible( showLabelOptions );
-  mLabelOptions->setDialogTitle( tr( "Customize label text" ) );
-  connect( mLabels, &QCheckBox::stateChanged, this, &QgsPointCloudRendererPropertiesWidget::emitWidgetChanged );
-  connect( mLabelOptions, &QgsFontButton::changed, this, &QgsPointCloudRendererPropertiesWidget::emitWidgetChanged );
+  // show virtual point cloud options only when vpc layer is selected
+  if ( !mLayer->dataProvider()->subIndexes().isEmpty() )
+  {
+    mLabelOptions->setDialogTitle( tr( "Customize label text" ) );
+    connect( mLabels, &QCheckBox::stateChanged, this, &QgsPointCloudRendererPropertiesWidget::emitWidgetChanged );
+    connect( mLabelOptions, &QgsFontButton::changed, this, &QgsPointCloudRendererPropertiesWidget::emitWidgetChanged );
+    mZoomOutOptions->addItem( tr( "Show extents only" ), static_cast<int>( Qgis::PointCloudZoomOutRenderBehavior::RenderExtents ) );
+
+    if ( const QgsVirtualPointCloudProvider *vpcProvider = dynamic_cast<QgsVirtualPointCloudProvider *>( mLayer->dataProvider() ) )
+    {
+      if ( vpcProvider->overview() )
+      {
+        mZoomOutOptions->addItem( tr( "Show overview only" ), static_cast<int>( Qgis::PointCloudZoomOutRenderBehavior::RenderOverview ) );
+        mZoomOutOptions->addItem( tr( "Show extents over overview" ), static_cast<int>( Qgis::PointCloudZoomOutRenderBehavior::RenderOverviewAndExtents ) );
+      }
+    }
+    else
+    {
+      mZoomOutOptions->setEnabled( false );
+    }
+
+    connect( mZoomOutOptions, qOverload<int>( &QComboBox::currentIndexChanged ), this, [this]( int currentZoomOutOption ) {
+      switch ( static_cast<Qgis::PointCloudZoomOutRenderBehavior>( currentZoomOutOption ) )
+      {
+        case Qgis::PointCloudZoomOutRenderBehavior::RenderOverview:
+          mLabels->setEnabled( false );
+          mLabelOptions->setEnabled( false );
+          break;
+        default:
```

Comment:
Avoid default when switching on enum values, so we get a build time warning if more enum values are added in future that need to be handled here